{
	"document": {
		"aggregate_severity": {
			"namespace": "https://access.redhat.com/security/updates/classification/",
			"text": "Important"
		},
		"category": "csaf_vex",
		"csaf_version": "2.0",
		"distribution": {
			"text": "Copyright Â© Red Hat, Inc. All rights reserved.",
			"tlp": {
				"label": "WHITE",
				"url": "https://www.first.org/tlp/"
			}
		},
		"lang": "en",
		"notes": [
			{
				"category": "summary",
				"text": "Updated python-keystoneclient packages that fix one security issue are now\navailable for Red Hat Enterprise Linux OpenStack Platform 3.0.\n\nThe Red Hat Security Response Team has rated this update as having\nImportant security impact. A Common Vulnerability Scoring System (CVSS)\nbase score, which gives a detailed severity rating, is available from the\nCVE link in the References section.",
				"title": "Topic"
			},
			{
				"category": "general",
				"text": "Python-keystoneclient is a client library and a command line utility for\ninteracting with the OpenStack Identity API. The OpenStack Identity\nauth_token middleware component handles the authentication of tokens\nwith keystone.\n\nWhen using the auth_token middleware with the memcached token cache\nenabled, a token for a different identity could be returned. An\nauthenticated user could use this flaw to escalate their privileges by\nmaking repeated requests that could eventually allow the user to acquire\nthe administrator's identity. Note that only OpenStack Identity setups\nusing auth_token with memcached were affected. (CVE-2014-0105)\n\nRed Hat would like to thank the OpenStack project for reporting this issue.\nUpstream acknowledges Kieran Spear from the University of Melbourne as the\noriginal reporter.\n\nAll python-keystoneclient users are advised to upgrade to these updated\npackages, which correct this issue. After installing this update, all\nOpenStack services using auth_token must be restarted for this update to\ntake effect.",
				"title": "Details"
			},
			{
				"category": "legal_disclaimer",
				"text": "This content is licensed under the Creative Commons Attribution 4.0 International License (https://creativecommons.org/licenses/by/4.0/). If you distribute this content, or a modified version of it, you must provide attribution to Red Hat Inc. and provide a link to the original.",
				"title": "Terms of Use"
			}
		],
		"publisher": {
			"category": "vendor",
			"contact_details": "https://access.redhat.com/security/team/contact/",
			"issuing_authority": "Red Hat Product Security is responsible for vulnerability handling across all Red Hat offerings.",
			"name": "Red Hat Product Security",
			"namespace": "https://www.redhat.com"
		},
		"references": [
			{
				"category": "self",
				"summary": "https://access.redhat.com/errata/RHSA-2014:0442",
				"url": "https://access.redhat.com/errata/RHSA-2014:0442"
			},
			{
				"category": "external",
				"summary": "https://access.redhat.com/security/updates/classification/#important",
				"url": "https://access.redhat.com/security/updates/classification/#important"
			},
			{
				"category": "external",
				"summary": "1082165",
				"url": "https://bugzilla.redhat.com/show_bug.cgi?id=1082165"
			},
			{
				"category": "self",
				"summary": "Canonical URL",
				"url": "https://access.redhat.com/security/data/csaf/v2/advisories/2014/rhsa-2014_0442.json"
			}
		],
		"title": "Red Hat Security Advisory: python-keystoneclient security update",
		"tracking": {
			"current_release_date": "2024-08-14T13:28:53+00:00",
			"generator": {
				"date": "2024-08-14T13:28:53+00:00",
				"engine": {
					"name": "Red Hat SDEngine",
					"version": "3.32.0"
				}
			},
			"id": "RHSA-2014:0442",
			"initial_release_date": "2014-04-28T20:06:16+00:00",
			"revision_history": [
				{
					"date": "2014-04-28T20:06:16+00:00",
					"number": "1",
					"summary": "Initial version"
				},
				{
					"date": "2014-04-28T20:06:17+00:00",
					"number": "2",
					"summary": "Last updated version"
				},
				{
					"date": "2024-08-14T13:28:53+00:00",
					"number": "3",
					"summary": "Last generated version"
				}
			],
			"status": "final",
			"version": "3"
		}
	},
	"product_tree": {
		"branches": [
			{
				"branches": [
					{
						"branches": [
							{
								"category": "product_name",
								"name": "Red Hat Enterprise Linux OpenStack Platform 3.0",
								"product": {
									"name": "Red Hat Enterprise Linux OpenStack Platform 3.0",
									"product_id": "6Server-Grizzly",
									"product_identification_helper": {
										"cpe": "cpe:/a:redhat:openstack:3::el6"
									}
								}
							}
						],
						"category": "product_family",
						"name": "Red Hat OpenStack Platform"
					},
					{
						"branches": [
							{
								"category": "product_version",
								"name": "python-keystoneclient-doc-1:0.2.3-8.el6ost.noarch",
								"product": {
									"name": "python-keystoneclient-doc-1:0.2.3-8.el6ost.noarch",
									"product_id": "python-keystoneclient-doc-1:0.2.3-8.el6ost.noarch",
									"product_identification_helper": {
										"purl": "pkg:rpm/redhat/python-keystoneclient-doc@0.2.3-8.el6ost?arch=noarch&epoch=1"
									}
								}
							},
							{
								"category": "product_version",
								"name": "python-keystoneclient-1:0.2.3-8.el6ost.noarch",
								"product": {
									"name": "python-keystoneclient-1:0.2.3-8.el6ost.noarch",
									"product_id": "python-keystoneclient-1:0.2.3-8.el6ost.noarch",
									"product_identification_helper": {
										"purl": "pkg:rpm/redhat/python-keystoneclient@0.2.3-8.el6ost?arch=noarch&epoch=1"
									}
								}
							}
						],
						"category": "architecture",
						"name": "noarch"
					},
					{
						"branches": [
							{
								"category": "product_version",
								"name": "python-keystoneclient-1:0.2.3-8.el6ost.src",
								"product": {
									"name": "python-keystoneclient-1:0.2.3-8.el6ost.src",
									"product_id": "python-keystoneclient-1:0.2.3-8.el6ost.src",
									"product_identification_helper": {
										"purl": "pkg:rpm/redhat/python-keystoneclient@0.2.3-8.el6ost?arch=src&epoch=1"
									}
								}
							}
						],
						"category": "architecture",
						"name": "src"
					}
				],
				"category": "vendor",
				"name": "Red Hat"
			}
		],
		"relationships": [
			{
				"category": "default_component_of",
				"full_product_name": {
					"name": "python-keystoneclient-1:0.2.3-8.el6ost.noarch as a component of Red Hat Enterprise Linux OpenStack Platform 3.0",
					"product_id": "6Server-Grizzly:python-keystoneclient-1:0.2.3-8.el6ost.noarch"
				},
				"product_reference": "python-keystoneclient-1:0.2.3-8.el6ost.noarch",
				"relates_to_product_reference": "6Server-Grizzly"
			},
			{
				"category": "default_component_of",
				"full_product_name": {
					"name": "python-keystoneclient-1:0.2.3-8.el6ost.src as a component of Red Hat Enterprise Linux OpenStack Platform 3.0",
					"product_id": "6Server-Grizzly:python-keystoneclient-1:0.2.3-8.el6ost.src"
				},
				"product_reference": "python-keystoneclient-1:0.2.3-8.el6ost.src",
				"relates_to_product_reference": "6Server-Grizzly"
			},
			{
				"category": "default_component_of",
				"full_product_name": {
					"name": "python-keystoneclient-doc-1:0.2.3-8.el6ost.noarch as a component of Red Hat Enterprise Linux OpenStack Platform 3.0",
					"product_id": "6Server-Grizzly:python-keystoneclient-doc-1:0.2.3-8.el6ost.noarch"
				},
				"product_reference": "python-keystoneclient-doc-1:0.2.3-8.el6ost.noarch",
				"relates_to_product_reference": "6Server-Grizzly"
			}
		]
	},
	"vulnerabilities": [
		{
			"acknowledgments": [
				{
					"names": [
						"OpenStack project"
					]
				},
				{
					"names": [
						"Kieran Spear"
					],
					"organization": "University of Melbourne",
					"summary": "Acknowledged by upstream."
				}
			],
			"cve": "CVE-2014-0105",
			"discovery_date": "2014-03-24T00:00:00+00:00",
			"ids": [
				{
					"system_name": "Red Hat Bugzilla ID",
					"text": "1082165"
				}
			],
			"notes": [
				{
					"category": "description",
					"text": "The auth_token middleware in the OpenStack Python client library for Keystone (aka python-keystoneclient) before 0.7.0 does not properly retrieve user tokens from memcache, which allows remote authenticated users to gain privileges in opportunistic circumstances via a large number of requests, related to an \"interaction between eventlet and python-memcached.\"",
					"title": "Vulnerability description"
				},
				{
					"category": "summary",
					"text": "python-keystoneclient: Potential context confusion in Keystone middleware",
					"title": "Vulnerability summary"
				},
				{
					"category": "general",
					"text": "The CVSS score(s) listed for this vulnerability do not reflect the associated product's status, and are included for informational purposes to better understand the severity of this vulnerability.",
					"title": "CVSS score applicability"
				}
			],
			"product_status": {
				"fixed": [
					"6Server-Grizzly:python-keystoneclient-1:0.2.3-8.el6ost.noarch",
					"6Server-Grizzly:python-keystoneclient-1:0.2.3-8.el6ost.src",
					"6Server-Grizzly:python-keystoneclient-doc-1:0.2.3-8.el6ost.noarch"
				]
			},
			"references": [
				{
					"category": "self",
					"summary": "Canonical URL",
					"url": "https://access.redhat.com/security/cve/CVE-2014-0105"
				},
				{
					"category": "external",
					"summary": "RHBZ#1082165",
					"url": "https://bugzilla.redhat.com/show_bug.cgi?id=1082165"
				},
				{
					"category": "external",
					"summary": "https://www.cve.org/CVERecord?id=CVE-2014-0105",
					"url": "https://www.cve.org/CVERecord?id=CVE-2014-0105"
				},
				{
					"category": "external",
					"summary": "https://nvd.nist.gov/vuln/detail/CVE-2014-0105",
					"url": "https://nvd.nist.gov/vuln/detail/CVE-2014-0105"
				}
			],
			"release_date": "2014-03-27T00:00:00+00:00",
			"remediations": [
				{
					"category": "vendor_fix",
					"details": "Before applying this update, make sure all previously released errata\nrelevant to your system have been applied.\n\nThis update is available via the Red Hat Network. Details on how to\nuse the Red Hat Network to apply this update are available at\nhttps://access.redhat.com/site/articles/11258",
					"product_ids": [
						"6Server-Grizzly:python-keystoneclient-1:0.2.3-8.el6ost.noarch",
						"6Server-Grizzly:python-keystoneclient-1:0.2.3-8.el6ost.src",
						"6Server-Grizzly:python-keystoneclient-doc-1:0.2.3-8.el6ost.noarch"
					],
					"restart_required": {
						"category": "none"
					},
					"url": "https://access.redhat.com/errata/RHSA-2014:0442"
				}
			],
			"scores": [
				{
					"cvss_v2": {
						"accessComplexity": "MEDIUM",
						"accessVector": "NETWORK",
						"authentication": "NONE",
						"availabilityImpact": "NONE",
						"baseScore": 4.3,
						"confidentialityImpact": "NONE",
						"integrityImpact": "PARTIAL",
						"vectorString": "AV:N/AC:M/Au:N/C:N/I:P/A:N",
						"version": "2.0"
					},
					"products": [
						"6Server-Grizzly:python-keystoneclient-1:0.2.3-8.el6ost.noarch",
						"6Server-Grizzly:python-keystoneclient-1:0.2.3-8.el6ost.src",
						"6Server-Grizzly:python-keystoneclient-doc-1:0.2.3-8.el6ost.noarch"
					]
				}
			],
			"threats": [
				{
					"category": "impact",
					"details": "Important"
				}
			],
			"title": "python-keystoneclient: Potential context confusion in Keystone middleware"
		}
	]
}
